lib/Runtime/Types/TypePath.h

```diff
@@ -8,7 +8,7 @@ namespace Js
 {
     class TinyDictionary
     {
-        static const int PowerOf2_BUCKETS = 8;
+        static const int PowerOf2_BUCKETS = 16;
```

Comment:
The change widens TinyDictionary to 16 buckets without changing its size or capacity. It looks like the highest value bit is unused ans can be used as a sentinel - that buys extra space with the same table size.

Current situation with 8 buckets and up to 128 items capacity can result in a lot of chaining and that hurts querying performance. With 16 buckets the chains should be 1/2 shorter on average.

Reply:
Also when augmented values act as sentinels, we may not need to lookup in `bucketsData` at all on sub-16 type paths. 
That should be helpful too.

Reply:
The change seems fine. Have you run it against benchmarks (Octane, Speedometer, etc.)?

Reply:
I have a repro case for a SCA deserialization. It heavily uses this dictionary since it basically adds numerous properties one by one. The change results in an improvement. 
It is unlikely there is a scenario where more buckets would be a negative negative change. 

Running the benchmarks would be interesting.

Reply:
Hmm. I see about 10% regression in either benchmark. Will need to figure why before merging. 

The only thing I can think of right now is the change in string hashing. Even though the old hash is not very good, it might work really well for very short strings - "a", "b", "c"

Reply:
After retrying this I no longer see the difference in benchmarks. If I collect results from two _patched_ instances - one with the changes and one with baseline bits, the results are roughly the same.
I think I compared results from patched and unpatched, and that is not exactly apple-to-aple comparison.